packages/oats-runtime/src/make.ts

```diff
@@ -246,15 +272,20 @@ export function makeOptional(maker: any) {
   return { optional: maker };
 }
 
-function checkBoolean(value: any) {
-  if (typeof value !== 'boolean') {
-    return getErrorWithValueMsg('expected a boolean', value);
-  }
-  return Make.ok(value);
-}
-
-export function makeBoolean() {
-  return checkBoolean;
+export function makeBoolean(): Maker<unknown, boolean> {
+  return (x: unknown, { parseBooleanStrings = false } = {}) => {
+    let value: boolean;
+    if (typeof x === 'boolean') {
+      value = x;
+    } else if (parseBooleanStrings && x === 'true') {
+      value = true;
+    } else if (parseBooleanStrings && x === 'false') {
+      value = false;
+    } else {
+      return getErrorWithValueMsg('expected a boolean', x as boolean);
```

Comment:
does boolean check have to be this specific? Ideally I would prefer anything that is not `false | null | undefined` be evaluated as `true`

Reply:
so `"gibberish"` will be considered as `true`?

Reply:
the original implementation required value to be of boolean type, so it didn't allow null, undefined etc.

Reply:
if we would map `undefined` to `false` how would optional boolean values be handled? To me not providing a value is different to setting it to `false`.

I other words I would go with the strictest possible interpretation

Reply:
Its just an opinion, but I strongly feel that the result should be evaluated by the expression `Boolean(value)`. So `Boolean('gibberish')` will be true and even `Boolean('false')` will also be true. If one really wants a false, then the value can be anything that evaluates to false like `'' | false | null | 0 `

Reply:
I'd go for strictness too. Most times, following strict rules is easy, debugging loosely written code is hard.

Reply:
@sakari a boolean `undefined` does evaluate to `false`, I don't see any other way an optional boolean undefined should be interpreted. I do agree with strictness when its necessary but its also a balance that code does not end up being rigid to changes. However, I am unaware of the full context, so I concede to what you guys feel is more feasible
```
if (shouldDoSomething && shouldDoSomething === true) {
  // do something
} else {
  // don't do something
}
```

Reply:
not having a property is different to having a property eg in cases where you try to transmit a delta with some homegrown framework (ofc a questionable idea in general)

Reply:
Might be worth noting that OpenAPI spcs are not a JavaScript thing, so using JavaScript concepts to determine what is `false` and `true` might lead to APIs implemented in other languages behaving differently when their clients are implemented in JavaScript (or vice versa)

Not every language agrees that `''` is falsey, but every language agrees that `false` is `false`.